lib/thesis/controller.ex

```diff
@@ -43,6 +90,7 @@ defmodule Thesis.Controller.Plug do
     conn
     |> assign(:thesis_page, current_page)
     |> assign(:thesis_content, page_contents)
+    |> assign(:thesis_editable, auth.page_is_editable?(conn))
   end
```

Comment:
For here, I'm caching the current page, any page contents, and whether the current user has the ability to edit this page. This gets attached (as an `assign`) to the `conn`. You can access them in a view with `@thesis_page` or `@thesis_editable`, etc, although you usually don't need to; they're more for Thesis's internal benefit.